### ISSUE TITLE
Suppress only-non-binary-in-usr-lib if there are symlinks to binaries in lib dir

### ISSUE THREAD

**Issue by scop** [rpm-software-management/rpmlint#73]
For example unversioned `*.so` links to versioned binary `*.so.x.y` which are common in devel packages. Either just do this for all such symlinks, or actually examine the target of the symlink to see if it's binary. The latter is cumbersome and would only work for installed packages, while the former might be a bit broad. See https://bugzilla.redhat.com/show_bug.cgi?id=1330804

**Comment by StefanBruens** [comment]
This issue is still present with current rpmlint, though the message has slightly changed.

I think the severity of this issue should be raised, as:  
1. This is an error now
2. It happens for every package which has a devel package

Example rpmplint output:
```
gpsnmea-devel.x86_64: E: no-binary
quat-devel.x86_64: E: no-binary
The package should be of the noarch architecture because it doesn't contain
any binaries.
```

**Comment by StefanBruens** [comment]
The relevant code is here:

https://github.com/rpm-software-management/rpmlint/blob/3c828527f0e1435dd3779d5e47d7be5698929511/rpmlint/checks/BinariesCheck.py#L634-L635

I think the `ISREG` is wrong for two reasons:  
1. It triggers the warning here
2. It fails to detect any directory or symlink in `/usr/lib64` and allows it in a noarch package.

Installing such a noarch package would install stuff in `/usr/lib64` even on a 32 bit platform.

**Comment by marxin** [comment]
I agree with your analysis and I've created a pull request. Feel free to comment there.
